src/forest.jl

```diff
@@ -0,0 +1,52 @@
+mutable struct Forest{T<:Integer}
+    counter::T
+    intmap::Dict{Tuple{T,T},T}
+    parents::Vector{T}
+    ranks::Vector{T}
+    ntrees::T
+end
+
+function Forest{T}(n::Integer) where {T<:Integer}
+    counter = zero(T)
+    intmap = Dict{Tuple{T,T},T}()
+    sizehint!(intmap, n)
+    parents = collect(Base.OneTo(T(n)))
+    ranks = zeros(T, T(n))
+    ntrees = zero(T)
+    return Forest{T}(counter, intmap, parents, ranks, ntrees)
+end
+
+function Base.push!(forest::Forest{T}, edge::Tuple{T,T}) where {T<:Integer}
+    forest.counter += 1
+    forest.intmap[edge] = forest.counter
+    forest.ntrees += one(T)
+    return edge
```

Comment:
```suggestion
    return forest
```
typically, modifying methods return the object they're modifying